application/org.openjdk.jmc.ui.common/src/main/java/org/openjdk/jmc/ui/common/security/PersistentCredentials.java

```diff
@@ -78,4 +81,12 @@ private String[] getCredentials() throws SecurityException {
 	public String getExportedId() {
 		return id;
 	}
+
+	public static boolean isPasswordValid(final String password) {
+		Pattern PASSWORD_PATTERN = Pattern
+				.compile("^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#(&)[{-}]:;',?/*~$^+=<>]).{8,20}$"); //$NON-NLS-1$
```

Comment:
I think we could make this quite a lot simpler. Generally I'd say that password length is more important than checking for special characters and numbers.